inc/azure_macro_utils/macro_utils.h

```diff
@@ -239,17 +237,27 @@ const char* MU_C3(MU_, enumIdentifier,_ToString)(enumIdentifier value)
 #define CONSTRUCT_FROM_FAKE_ENUM(from_value, to_value) \
     MU_C3(fake_from_, from_value, _C0FB3709_EDE8_4288_8F70_FDDB5D8D7A51),
 
+#define CONSTRUCT_FIELD_FROM_FAKE_ENUM(from_value, to_value) \
+    int MU_C3(fake_from_, from_value, _C0FB3709_EDE8_4288_8F70_FDDB5D8D7A51)[(MU_C2(enum_value_typedef_, from_value) == MU_C2(enum_value_typedef_, from_value)) ? 1 : 0];
+
+//    int MU_C3(fake_from_, from_value, _C0FB3709_EDE8_4288_8F70_FDDB5D8D7A51)[(MU_C2(enum_value_typedef_, from_value) == MU_C2(enum_value_typedef_, from_value)) ? 1 : 0];
+
 // This macro creates a typedef for a from enum value
 // Its type is the enum value of the original enum in order to make sure that only valid
 // enum values are used in from
 #define MU_TYPEDEF_FROM_ENUM_VALUE(from_value, to_value) \
-    typedef MU_C2(enum_value_typedef_, from_value) MU_C2(enum_value_typedef_convert_from_, from_value);
+    MU_C2(enum_value_typedef_convert_from_, from_value);
 
 // This macro adds validation of the fact that the from values count in the list match the count of values in the
 // (as opposed to the macro MU_DEFINE_CONVERT_ENUM_WITHOUT_VALIDATION)
 // from enum type. It does that by having a special fake enum where all the "from" values are enumerated
 // The count check is done by having a zero sized array in a struct if the counts do not match
 #define MU_DEFINE_CONVERT_ENUM(ENUM_TYPE_FROM, ENUM_TYPE_TO, ...) \
+    /* This forces matching the values in from to enum values generated with MU_DEFINE_ENUM */ \
+    typedef struct MU_C3(fake_check_enum_values_exist_, ENUM_TYPE_FROM, _C0FB3709_EDE8_4288_8F70_FDDB5D8D7A51_TAG) \
```

Comment:
>typedef [](start = 4, length = 7)

While I'm thinking of it, it might be nice to also validate the "to" enums, avoid typos like:

```
MU_DEFINE_CONVERT_ENUM(TEST_CONVERT_FROM_ENUM, TEST_CONVERT_TO_ENUM,
    test_from_a, test_from_a,
    test_from_b, test_to_b,
    test_from_c, test_to_b);
```

Just another task to consider

Reply:
Yeah, I left this on purpose because I thought we need to translate several from enum values to the same to enum value.
What might be nice as a separate task would be to make sure all the to enum values are used for example (even if one is used twice).

---
In reply to: [396528535](https://github.com/Azure/azure-macro-utils-c/pull/37#discussion_r396528535) [](ancestors = 396528535)